test/e2e/authorization.go

```diff
@@ -0,0 +1,572 @@
+				tc.run(context.TODO(), cs, false)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+			g.It("should allow access to resources other than Nodes", func() {
+				tc.data.resources = []string{
+					"namespaces",
+					"storage.k8s.io/storageclasses",
+					"apiextensions.k8s.io/customresourcedefinitions",
+				}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+		})
+	})
+
+	g.Context("For system users", func() {
+		var tc testCase
+
+		g.When("the user is kubelet", func() {
+			g.BeforeEach(func() {
+				tc.data.groups = [][]string{{"system:masters"}}
+				tc.data.users = []string{"kubelet"}
+			})
+			g.It("should allow to get Pods", func() {
+				tc.data.resources = []string{"pods"}
+				tc.data.verbs = []string{"get"}
+				tc.data.namespaces = []string{"teapot"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+		})
+
+		g.When("the service account is daemonset-controller", func() {
+			g.BeforeEach(func() {
+				tc.data.users = []string{"system:serviceaccount:kube-system:daemon-set-controller"}
+				tc.data.groups = [][]string{{"system:serviceaccounts:kube-system"}}
+			})
+			g.It("should allow to update DaemonSet status subresource", func() {
+				tc.data.resources = []string{"apps/daemonsets/status"}
+				tc.data.verbs = []string{"update"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+			g.It("should allow to update DaemonSet finalizers", func() {
+				tc.data.resources = []string{"apps/daemonsets/finalizers"}
+				tc.data.verbs = []string{"update"}
+				tc.run(context.TODO(), cs, true)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+			// TODO: Need to verify in the original tests if this is a permission on
+			// the controller-manager or the daemonset-controller.
+			// g.It("should allow to create Pods", func() {})
+		})
+
+		g.When("the default service account is in default namespace", func() {
+			g.BeforeEach(func() {
+				tc.data.users = []string{"system:serviceaccount:default:default"}
+			})
+			g.It("should deny to list StatefulSets", func() {
+				tc.data.resources = []string{"apps/statefulsets"}
+				tc.data.verbs = []string{"list"}
+				tc.run(context.TODO(), cs, false)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+			})
+		})
+
+		g.When("the default service account is in non-default namespace", func() {
+			g.BeforeEach(func() {
+				tc.data.users = []string{"system:serviceaccount:non-default:default"}
+			})
+			g.It("should deny to list StatefulSets", func() {
+				tc.data.resources = []string{"apps/statefulsets"}
+				tc.data.verbs = []string{"list"}
+				tc.run(context.TODO(), cs, false)
+				gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
```

Comment:
This is the same behaviour as the previous test, right? Can we combine them?

Reply:
Or just one test with this?

```
tc.data.users = []string{"system:serviceaccount:default:default", "system:serviceaccount:non-default:default"}
```

Reply:
Ah yes. you're right. They're identical 😅 I was focussing too much on copying the tests 1 to 1. I merged them into one test case specifying both users.

Combined them in one test in: https://github.com/zalando-incubator/kubernetes-on-aws/pull/8536/commits/4eba59b8cb63f1d56b7ef05fad86022bfe76ebae